File: sundries/ctf/2020/decompetition/solutions/baby-go/source.go
// baby-go reconstruction
// by Aleksey
// - https://alekseyyy.github.io
// - @EpsilonCalculus / @EntropyThot

package main

import (
  "fmt"
  "os"
  "strconv"
)

func main() {
  // Good luck!
  
  if (len(os.Args) > 1) {
    input, err := strconv.Atoi(os.Args[1])
    if err == nil {
      if (input % 3 == 0 && input % 5 == 0) {
        fmt.Println("Green")
      } else if (input % 3 == 0) {
        fmt.Println("Yellow")
      } else if (input % 5 == 0) {
        fmt.Println("Blue")
      } else {
        fmt.Println(input)
      }
    }
  }
}

File: sundries/ctf/2020/decompetition/solutions/baby-c/source.c
/*
 * baby-c reconstruction
 * by Aleksey (mostly with help from Ghidra's recompiler :P)
 * - https://alekseyyy.github.io
 * - @EpsilonCalculus / @EntropyThot
*/

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

long foo(int a, int b) {

  int x, y;
  x = b;
  y = a;
  
  while (x != y) {
    if (x < y)
      y = y - x;
    else {
      if (y < x)
        x = x - y;
    }
  }
  
  return y;
}

int main(int argc, char** argv) {
  
  int x1, x2, y;
  
  if (argc != 3) {
    fprintf(stderr, "USAGE: %s <num1> <num2>\n", *argv);
    exit(1);
  }
  
  x1 = atoi((char*)argv[1]);
  x2 = atoi((char*)argv[2]);
  y = foo(x1, x2);
  printf("foo(%d, %d) = %d\n", x1, x2, y);
  return 0;
}
